conn_test.go

```diff
@@ -67,6 +67,21 @@ func TestValidCommunication(t *testing.T) {
 		t.Fatalf("unexpected local address: %v", addr)
 	}
 
+	// test against regression from https://github.com/pion/mdns/commit/608f20b
+	// where by properly sending mDNS responses to all interfaces, we significantly
+	// increased the chance that we send a loopback response to a Query that is
+	// unwillingly to use loopback addresses (the default in pion/ice).
+	for i := 0; i < 100; i++ {
+		_, addr, err = bServer.Query(context.TODO(), "pion-mdns-2.local")
```

Comment:
This should prevent future mishaps but it's racy. From what we know on https://github.com/pion/webrtc/issues/2625, it easily fails within 10 attempts since loopback has a higher chance of routing back to itself faster than a LAN mDNS response (I think)